packages/opentelemetry-tracing/src/BasicTracerFactory.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { BasicTracer } from './BasicTracer';
+
+export class BasicTracerFactory implements types.TracerFactory {
+  /** Singleton instance */
+  private static _singletonInstance: types.TracerFactory;
+  private readonly _tracers: Map<String, BasicTracer> = new Map();
+
+  getTracer(name: string = '', version?: string): types.Tracer {
+    const key = name + (version != undefined ? version : '');
+    if (this._tracers.has(key)) return this._tracers.get(key)!;
+
+    const tracer = new BasicTracer();
+    this._tracers.set(key, tracer);
+    return tracer;
+  }
+
+  // @todo
+  // addSpanProcessor(spanProcessor: SpanProcessor): void { }
```

Comment:
I take it that this would be applied to all tracers in the set and to all new tracers created afterwards?

Reply:
Yes, correct.